### ISSUE TITLE
Integration issue with Angular-CLI's new Webpack version

### ISSUE THREAD

**Issue by CalvinDale** [angular-redux/ng2-redux#180]
I'm trying to integrate the `ng2-redux` library with a new project created via the latest version of the `angular-cli` which now uses _Webpack_ rather than _System.js_.

However, I'm encountering some warnings which behave as errors since they prevent the library from working. Here are the steps to reproduce:

1. `> npm install -g angular-cli@webpack --no-optional`
2. `> ng new {ProjectName}`
3. `> cd {ProjectName}`
4. In `package.json`, remove the circumflex in the `"angular-cli": "1.0.0-beta.11-webpack"` reference
5. `> npm i --no-optional` 
6. `> npm i redux ng2-redux --no-optional`
7. `ng serve` (should run now)
8. Edit `main.ts` to incorporate `ng2-redux`:
```
import { NgRedux } from 'ng2-redux';
import { bootstrap } from '@angular/platform-browser-dynamic';
import { enableProdMode } from '@angular/core';
import { AppComponent, environment } from './app/';

if (environment.production) {
  enableProdMode();
}

bootstrap(AppComponent, [NgRedux]);
```
Run `ng serve` again. The following warnings now appear in shell and browser consoles:
```
WARNING in ./~/ng2-redux/lib/index.js
Cannot find source file '../src/index.ts': Error: Can't resolve '../src/index.ts' in '{ProjectPath\Name}\node_modules\ng2-redux\lib'
 @ ./src/main.ts 1:0-36
 @ multi main

WARNING in ./~/ng2-redux/lib/components/ng-redux.js
Cannot find source file '../../src/components/ng-redux.ts': Error: Can't resolve '../../src/components/ng-redux.ts' in '{ProjectPath\Name}\node_modules\ng2-redux\lib\components'
 @ ./~/ng2-redux/lib/index.js 2:17-49
 @ ./src/main.ts
 @ multi main

WARNING in ./~/ng2-redux/lib/components/dev-tools.js
Cannot find source file '../../src/components/dev-tools.ts': Error: Can't resolve '../../src/components/dev-tools.ts' in '{ProjectPath\Name}\node_modules\ng2-redux\lib\components'
 @ ./~/ng2-redux/lib/index.js 4:18-51
 @ ./src/main.ts
 @ multi main

... SEVERAL OTHERS ...
```
And the `ng2-redux` functionality doesn't work when a simple counter example is wired up similar to the _Quick Start_.

Any ideas as to how to get this to work?

**Comment by SethDavenport** [comment]
Hmm - looks like webpack is trying to pull in the ts files; we only publish the JS files.

I'll have to take a look at what the new cli is doing.

**Comment by SethDavenport** [comment]
FWIW I've used ng2-redux with webpack (no CLI), and I've used it with the SystemJS CLI.  I'll see what the new CLI is setting up config-wise.

**Comment by SethDavenport** [comment]
Spent some time with the new webpack-based Angular-CLI. We have some adjusting to do.  Particularly:

* they're on webpack 2 beta
* they're on typescript 2 beta
* our peer dependency on typings probably shouldn't be there
* they seem to expect us to package our .ts files in npm; which we don't currently.

**Comment by SethDavenport** [comment]
@CalvinDale despite all this I was able to get a simple counter to work fine, even with the warnings. Here's a repo: https://github.com/SethDavenport/thing

**Comment by SethDavenport** [comment]
We'll look to see if including our .ts files in npm gets rid of the warnings.

**Comment by CalvinDale** [comment]
Thanks Seth! I tried out the `thing` project. The `package.json` doesn't include references to either `redux` or `ng2-redux`. And the project doesn't have a `counter` variable. So perhaps your subsequent work didn't get committed yet.

Also, I had to specify the exact reference to `angular-cli`.

`\thing [master ≡]> ng serve` reports:

```
No angular-cli-build.js found. Please see the transition guide: https://github.com/angular-cli/angular-cli/blob/master/TRANSITION.md#user-content-brocfile-transition.
Error: No angular-cli-build.js found. Please see the transition guide: https://github.com/angular-cli/angular-cli/blob/master/TRANSITION.md#user-content-brocfile-transition.
    at Class.module.exports.Task.extend.setupBroccoliBuilder (\thing\node_modules\angular-cli\lib\models\builder.js:57:13)
    at Class.module.exports.Task.extend.init (\thing\node_modules\angular-cli\lib\models\builder.js:89:10)
    at new Class (\thing\node_modules\core-object\core-object.js:18:12)
    at Class.module.exports.Task.extend.run (\thing\node_modules\angular-cli\lib\tasks\serve.js:15:19)
    at \thing\node_modules\angular-cli\lib\commands\serve.js:64:24
    at lib$rsvp$$internal$$tryCatch (\thing\node_modules\rsvp\dist\rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (\thing\node_modules\rsvp\dist\rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (\thing\node_modules\rsvp\dist\rsvp.js:1019:11)
    at lib$rsvp$asap$$flush (\thing\node_modules\rsvp\dist\rsvp.js:1198:9)
    at nextTickCallbackWith0Args (node.js:420:9)
\thing [master ≡]>
```

To fix, in `package.json` remove `^` in: `"angular-cli": "1.0.0-beta.11-webpack"` and update the install.

`\thing [master ≡]> npm i --no-optional`

And I had to manually delete the `\tmp` directory.

`\thing [master ≡]> ng serve` now reports:

```
\thing [master ≡ +1 ~1 -0 !]> ng serve
EEXIST: file already exists, mkdir '\thing\tmp'
Error: EEXIST: file already exists, mkdir '\thing\tmp'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:794:18)
    at testCanSymlink (\thing\node_modules\symlink-or-copy\index.js:18:6)
    at Object.<anonymous> (\thing\node_modules\symlink-or-copy\index.js:9:15)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
\thing [master ≡ +1 ~1 -0 !]>
```

Delete the `\thing\tmp\` directory and `ng serve` will work.

**Comment by SethDavenport** [comment]
Sorry that's what I get for coding at 1am - forgot to push!

It's updated now.

**Comment by CalvinDale** [comment]
I fetched the updated bits for the `thing` project. It's working now...so thanks for the demo. It's still reporting the aforementioned warnings on every modification during the subsequent Webpack compilation. It would be nice to get rid of that noise. I haven't had time yet to compare your project with mine to uncover the source my Webpack build failure.

**Comment by SethDavenport** [comment]
@CalvinDale The webpack warnings are fixed in 3.3.4.

**Comment by CalvinDale** [comment]
Perfect! I'll give it a try.
